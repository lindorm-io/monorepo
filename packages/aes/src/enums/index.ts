export * from "./AesEncryption";
export * from "./AesEncryptionKeyAlgorithm";
export * from "./AesFormat";
export * from "./AesIntegrityHash";
