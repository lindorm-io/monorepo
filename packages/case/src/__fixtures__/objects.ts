export const TEST_OBJECT = {
  camelCase: true,
  "Capital Case": true,
  CONSTANT_CASE: true,
  "dot.case": true,
  "kebab-case": true,
  "lower case": true,
  PascalCase: true,
  "path/case": true,
  "Sentence case": true,
  snake_case: true,
};

export const TEST_ARRAY_WITH_OBJECTS = [TEST_OBJECT, TEST_OBJECT];
