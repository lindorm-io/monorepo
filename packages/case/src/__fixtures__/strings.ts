export const TEST_STRINGS = [
  "camelCase",
  "Capital Case",
  "CONSTANT_CASE",
  "dot.case",
  "kebab-case",
  "lower case",
  "PascalCase",
  "path/case",
  "Sentence case",
  "snake_case",
];

export const INVALID_INPUT: Array<{ type: string; value: any }> = [
  { type: "number", value: 1234 },
  { type: "error", value: new Error("error") },
  { type: "date", value: new Date() },
  { type: "null", value: null },
  { type: "undefined", value: undefined },
];
