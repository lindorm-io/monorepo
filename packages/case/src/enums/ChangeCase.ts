export enum ChangeCase {
  Camel = "camel",
  Capital = "capital",
  Constant = "constant",
  Dot = "dot",
  Header = "header",
  Kebab = "kebab",
  Lower = "lower",
  Pascal = "pascal",
  Path = "path",
  Sentence = "sentence",
  Snake = "snake",
  None = "none",
}
