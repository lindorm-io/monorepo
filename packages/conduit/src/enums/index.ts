export * from "./ConduitUsing";
export * from "./ExpectedResponse";
