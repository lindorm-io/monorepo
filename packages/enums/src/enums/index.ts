export * from "./Environment";
export * from "./HttpMethod";
export * from "./PkceMethod";
