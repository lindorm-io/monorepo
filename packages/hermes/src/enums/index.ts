export * from "./ChecksumStoreType";
export * from "./EventStoreType";
export * from "./HermesStatus";
export * from "./MessageBusType";
export * from "./SagaStoreType";
export * from "./ViewStoreType";
