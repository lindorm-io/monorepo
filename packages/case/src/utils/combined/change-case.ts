import { ChangeCase } from "../../enums";
import {
  camelCase,
  capitalCase,
  constantCase,
  dotCase,
  headerCase,
  kebabCase,
  lowerCase,
  pascalCase,
  pathCase,
  sentenceCase,
  snakeCase,
} from "../specific";

export const changeCase = (input: string, mode: ChangeCase = ChangeCase.None): string => {
  switch (mode) {
    case ChangeCase.Camel:
      return camelCase(input);

    case ChangeCase.Capital:
      return capitalCase(input);

    case ChangeCase.Constant:
      return constantCase(input);

    case ChangeCase.Dot:
      return dotCase(input);

    case ChangeCase.Header:
      return headerCase(input);

    case ChangeCase.Kebab:
      return kebabCase(input);

    case ChangeCase.Lower:
      return lowerCase(input);

    case ChangeCase.Pascal:
      return pascalCase(input);

    case ChangeCase.Path:
      return pathCase(input);

    case ChangeCase.Sentence:
      return sentenceCase(input);

    case ChangeCase.Snake:
      return snakeCase(input);

    case ChangeCase.None:
      return input;

    default:
      throw new Error(`Invalid transform case [ ${mode} ]`);
  }
};
