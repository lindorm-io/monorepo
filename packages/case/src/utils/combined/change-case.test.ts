import { TEST_STRINGS } from "../../__fixtures__/strings";
import { ChangeCase } from "../../enums";
import { changeCase } from "./change-case";

describe.each([...Object.values(ChangeCase)])("changeCase with %s mode", (mode) => {
  test.each(TEST_STRINGS)("should convert %s", (input) => {
    expect(changeCase(input, mode)).toMatchSnapshot();
  });
});
