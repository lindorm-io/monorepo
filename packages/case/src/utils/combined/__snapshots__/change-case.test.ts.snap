// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changeCase with camel mode should convert CONSTANT_CASE 1`] = `"constantCase"`;

exports[`changeCase with camel mode should convert Capital Case 1`] = `"capitalCase"`;

exports[`changeCase with camel mode should convert PascalCase 1`] = `"pascalCase"`;

exports[`changeCase with camel mode should convert Sentence case 1`] = `"sentenceCase"`;

exports[`changeCase with camel mode should convert camelCase 1`] = `"camelCase"`;

exports[`changeCase with camel mode should convert dot.case 1`] = `"dotCase"`;

exports[`changeCase with camel mode should convert kebab-case 1`] = `"kebabCase"`;

exports[`changeCase with camel mode should convert lower case 1`] = `"lowerCase"`;

exports[`changeCase with camel mode should convert path/case 1`] = `"pathCase"`;

exports[`changeCase with camel mode should convert snake_case 1`] = `"snakeCase"`;

exports[`changeCase with capital mode should convert CONSTANT_CASE 1`] = `"Constant Case"`;

exports[`changeCase with capital mode should convert Capital Case 1`] = `"Capital Case"`;

exports[`changeCase with capital mode should convert PascalCase 1`] = `"Pascal Case"`;

exports[`changeCase with capital mode should convert Sentence case 1`] = `"Sentence Case"`;

exports[`changeCase with capital mode should convert camelCase 1`] = `"Camel Case"`;

exports[`changeCase with capital mode should convert dot.case 1`] = `"Dot Case"`;

exports[`changeCase with capital mode should convert kebab-case 1`] = `"Kebab Case"`;

exports[`changeCase with capital mode should convert lower case 1`] = `"Lower Case"`;

exports[`changeCase with capital mode should convert path/case 1`] = `"Path Case"`;

exports[`changeCase with capital mode should convert snake_case 1`] = `"Snake Case"`;

exports[`changeCase with constant mode should convert CONSTANT_CASE 1`] = `"CONSTANT_CASE"`;

exports[`changeCase with constant mode should convert Capital Case 1`] = `"CAPITAL_CASE"`;

exports[`changeCase with constant mode should convert PascalCase 1`] = `"PASCAL_CASE"`;

exports[`changeCase with constant mode should convert Sentence case 1`] = `"SENTENCE_CASE"`;

exports[`changeCase with constant mode should convert camelCase 1`] = `"CAMEL_CASE"`;

exports[`changeCase with constant mode should convert dot.case 1`] = `"DOT_CASE"`;

exports[`changeCase with constant mode should convert kebab-case 1`] = `"KEBAB_CASE"`;

exports[`changeCase with constant mode should convert lower case 1`] = `"LOWER_CASE"`;

exports[`changeCase with constant mode should convert path/case 1`] = `"PATH_CASE"`;

exports[`changeCase with constant mode should convert snake_case 1`] = `"SNAKE_CASE"`;

exports[`changeCase with dot mode should convert CONSTANT_CASE 1`] = `"constant.case"`;

exports[`changeCase with dot mode should convert Capital Case 1`] = `"capital.case"`;

exports[`changeCase with dot mode should convert PascalCase 1`] = `"pascal.case"`;

exports[`changeCase with dot mode should convert Sentence case 1`] = `"sentence.case"`;

exports[`changeCase with dot mode should convert camelCase 1`] = `"camel.case"`;

exports[`changeCase with dot mode should convert dot.case 1`] = `"dot.case"`;

exports[`changeCase with dot mode should convert kebab-case 1`] = `"kebab.case"`;

exports[`changeCase with dot mode should convert lower case 1`] = `"lower.case"`;

exports[`changeCase with dot mode should convert path/case 1`] = `"path.case"`;

exports[`changeCase with dot mode should convert snake_case 1`] = `"snake.case"`;

exports[`changeCase with header mode should convert CONSTANT_CASE 1`] = `"Constant-Case"`;

exports[`changeCase with header mode should convert Capital Case 1`] = `"Capital-Case"`;

exports[`changeCase with header mode should convert PascalCase 1`] = `"Pascal-Case"`;

exports[`changeCase with header mode should convert Sentence case 1`] = `"Sentence-Case"`;

exports[`changeCase with header mode should convert camelCase 1`] = `"Camel-Case"`;

exports[`changeCase with header mode should convert dot.case 1`] = `"Dot-Case"`;

exports[`changeCase with header mode should convert kebab-case 1`] = `"Kebab-Case"`;

exports[`changeCase with header mode should convert lower case 1`] = `"Lower-Case"`;

exports[`changeCase with header mode should convert path/case 1`] = `"Path-Case"`;

exports[`changeCase with header mode should convert snake_case 1`] = `"Snake-Case"`;

exports[`changeCase with kebab mode should convert CONSTANT_CASE 1`] = `"constant-case"`;

exports[`changeCase with kebab mode should convert Capital Case 1`] = `"capital-case"`;

exports[`changeCase with kebab mode should convert PascalCase 1`] = `"pascal-case"`;

exports[`changeCase with kebab mode should convert Sentence case 1`] = `"sentence-case"`;

exports[`changeCase with kebab mode should convert camelCase 1`] = `"camel-case"`;

exports[`changeCase with kebab mode should convert dot.case 1`] = `"dot-case"`;

exports[`changeCase with kebab mode should convert kebab-case 1`] = `"kebab-case"`;

exports[`changeCase with kebab mode should convert lower case 1`] = `"lower-case"`;

exports[`changeCase with kebab mode should convert path/case 1`] = `"path-case"`;

exports[`changeCase with kebab mode should convert snake_case 1`] = `"snake-case"`;

exports[`changeCase with lower mode should convert CONSTANT_CASE 1`] = `"constant case"`;

exports[`changeCase with lower mode should convert Capital Case 1`] = `"capital case"`;

exports[`changeCase with lower mode should convert PascalCase 1`] = `"pascal case"`;

exports[`changeCase with lower mode should convert Sentence case 1`] = `"sentence case"`;

exports[`changeCase with lower mode should convert camelCase 1`] = `"camel case"`;

exports[`changeCase with lower mode should convert dot.case 1`] = `"dot case"`;

exports[`changeCase with lower mode should convert kebab-case 1`] = `"kebab case"`;

exports[`changeCase with lower mode should convert lower case 1`] = `"lower case"`;

exports[`changeCase with lower mode should convert path/case 1`] = `"path case"`;

exports[`changeCase with lower mode should convert snake_case 1`] = `"snake case"`;

exports[`changeCase with none mode should convert CONSTANT_CASE 1`] = `"CONSTANT_CASE"`;

exports[`changeCase with none mode should convert Capital Case 1`] = `"Capital Case"`;

exports[`changeCase with none mode should convert PascalCase 1`] = `"PascalCase"`;

exports[`changeCase with none mode should convert Sentence case 1`] = `"Sentence case"`;

exports[`changeCase with none mode should convert camelCase 1`] = `"camelCase"`;

exports[`changeCase with none mode should convert dot.case 1`] = `"dot.case"`;

exports[`changeCase with none mode should convert kebab-case 1`] = `"kebab-case"`;

exports[`changeCase with none mode should convert lower case 1`] = `"lower case"`;

exports[`changeCase with none mode should convert path/case 1`] = `"path/case"`;

exports[`changeCase with none mode should convert snake_case 1`] = `"snake_case"`;

exports[`changeCase with pascal mode should convert CONSTANT_CASE 1`] = `"ConstantCase"`;

exports[`changeCase with pascal mode should convert Capital Case 1`] = `"CapitalCase"`;

exports[`changeCase with pascal mode should convert PascalCase 1`] = `"PascalCase"`;

exports[`changeCase with pascal mode should convert Sentence case 1`] = `"SentenceCase"`;

exports[`changeCase with pascal mode should convert camelCase 1`] = `"CamelCase"`;

exports[`changeCase with pascal mode should convert dot.case 1`] = `"DotCase"`;

exports[`changeCase with pascal mode should convert kebab-case 1`] = `"KebabCase"`;

exports[`changeCase with pascal mode should convert lower case 1`] = `"LowerCase"`;

exports[`changeCase with pascal mode should convert path/case 1`] = `"PathCase"`;

exports[`changeCase with pascal mode should convert snake_case 1`] = `"SnakeCase"`;

exports[`changeCase with path mode should convert CONSTANT_CASE 1`] = `"constant/case"`;

exports[`changeCase with path mode should convert Capital Case 1`] = `"capital/case"`;

exports[`changeCase with path mode should convert PascalCase 1`] = `"pascal/case"`;

exports[`changeCase with path mode should convert Sentence case 1`] = `"sentence/case"`;

exports[`changeCase with path mode should convert camelCase 1`] = `"camel/case"`;

exports[`changeCase with path mode should convert dot.case 1`] = `"dot/case"`;

exports[`changeCase with path mode should convert kebab-case 1`] = `"kebab/case"`;

exports[`changeCase with path mode should convert lower case 1`] = `"lower/case"`;

exports[`changeCase with path mode should convert path/case 1`] = `"path/case"`;

exports[`changeCase with path mode should convert snake_case 1`] = `"snake/case"`;

exports[`changeCase with sentence mode should convert CONSTANT_CASE 1`] = `"Constant case"`;

exports[`changeCase with sentence mode should convert Capital Case 1`] = `"Capital case"`;

exports[`changeCase with sentence mode should convert PascalCase 1`] = `"Pascal case"`;

exports[`changeCase with sentence mode should convert Sentence case 1`] = `"Sentence case"`;

exports[`changeCase with sentence mode should convert camelCase 1`] = `"Camel case"`;

exports[`changeCase with sentence mode should convert dot.case 1`] = `"Dot case"`;

exports[`changeCase with sentence mode should convert kebab-case 1`] = `"Kebab case"`;

exports[`changeCase with sentence mode should convert lower case 1`] = `"Lower case"`;

exports[`changeCase with sentence mode should convert path/case 1`] = `"Path case"`;

exports[`changeCase with sentence mode should convert snake_case 1`] = `"Snake case"`;

exports[`changeCase with snake mode should convert CONSTANT_CASE 1`] = `"constant_case"`;

exports[`changeCase with snake mode should convert Capital Case 1`] = `"capital_case"`;

exports[`changeCase with snake mode should convert PascalCase 1`] = `"pascal_case"`;

exports[`changeCase with snake mode should convert Sentence case 1`] = `"sentence_case"`;

exports[`changeCase with snake mode should convert camelCase 1`] = `"camel_case"`;

exports[`changeCase with snake mode should convert dot.case 1`] = `"dot_case"`;

exports[`changeCase with snake mode should convert kebab-case 1`] = `"kebab_case"`;

exports[`changeCase with snake mode should convert lower case 1`] = `"lower_case"`;

exports[`changeCase with snake mode should convert path/case 1`] = `"path_case"`;

exports[`changeCase with snake mode should convert snake_case 1`] = `"snake_case"`;
